universum/analyzers/utils.py

```diff
@@ -4,23 +4,27 @@
 import glob
 import subprocess
 
-from typing import Callable, List
+from typing import Callable, List, Optional
 from typing_extensions import TypedDict
 
 ReportData = TypedDict('ReportData', {'path': str, 'message': str, 'symbol': str, 'line': int})
 
 
-def report_parsed_outcome(cmd: List[str],
+def report_parsed_outcome(cmd: Optional[List[str]],
                           parse_function: Callable[[str], List[ReportData]],
                           out_file: str) -> int:
-    result = subprocess.run(cmd, universal_newlines=True,  # pylint: disable=subprocess-run-check
-                            stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    if result.stderr and not result.stdout:
-        sys.stderr.write(result.stderr)
-        return result.returncode
+    if cmd:
+        result = subprocess.run(cmd, universal_newlines=True,  # pylint: disable=subprocess-run-check
+                                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        if result.stderr and not result.stdout:
+            sys.stderr.write(result.stderr)
+            return result.returncode
+        data_to_parse = result.stdout
+    else:
+        data_to_parse = ''  # nothing to parse from stdout
 
     try:
-        parsed_issues = parse_function(result.stdout)
+        parsed_issues = parse_function(data_to_parse)
```

Comment:
If I understand the resulting logic of this function correctly it is like this:
1. normally run the specified command and call the function to parse
2. but in some cases instead don't run the command and pass the fake data to parse function.
The only reason I could imagine for this approach is that there is a possibility to reuse the single parsing function for both commandful and commandless scenarios.
Otherwise, it looks more reasonable to just pass the required data (parsed_issues) as an argument and wrap the "subprocess" part into another function to be used by clients. This would also remove the need for the callback.

Reply:
the code in question has been reworked